db/inmemory/client.go

```diff
@@ -43,17 +44,17 @@ func (c *client) AddUser(user *domain.User) (string, error) {
 	return user.ID, nil
 }
 
-func (c *client) ListUsers(limit int32, name string) ([]*domain.User, error) {
+func (c *client) ListUsers(limit int32, filteredMap map[string]interface{}) ([]*domain.User, error) {
 
 	var userInfo = []*domain.User{}
 	for _, value := range c.ds {
 		userInfo = append(userInfo, value)
 	}
 
 	userInfoName := make([]*domain.User, 0)
-	if name != "" {
+	if filteredMap["name"] != nil {
 		for i := 0; i < len(userInfo); i++ {
-			if name == userInfo[i].Name {
+			if strings.ToUpper(fmt.Sprint(filteredMap["name"])) == strings.ToUpper(userInfo[i].Name) {
```

Comment:
what is the purpose of toUpper here?

Reply:
Now when I will search the name in get request query parameter, it will search by converting the searched name & names stored in db to upper case letters. With the help of this, it does't matter, whether I am searching with upper case or lower case letters. If the name exists in db, it will Always provide me the request object.